app/routes/login-callback/route.tsx

```diff
@@ -0,0 +1,37 @@
+import { LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { commitSession, getSession, initializeEcomApiForRequest } from '~/lib/ecom/session';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+    const session = await getSession(request.headers.get('Cookie'));
+    const storedOauthData = session.get('oAuthData');
+
+    if (storedOauthData === undefined) {
+        return redirect('/login');
+    }
+
+    const api = await initializeEcomApiForRequest(request);
+    const wixClient = api.getWixClient();
+
+    const returnedOAuthData = wixClient.auth.parseFromUrl(request.url, 'query');
+    if (returnedOAuthData.error) {
+        throw new Error(`Error: ${returnedOAuthData.errorDescription}`);
+    }
+
+    const memberTokens = await wixClient.auth.getMemberTokens(
+        returnedOAuthData.code,
+        returnedOAuthData.state,
+        storedOauthData,
+    );
+    wixClient.auth.setTokens(memberTokens);
+
+    session.set('wixEcomTokens', memberTokens);
```

Comment:
Are these actually ecom tokens, or Wix API tokens in general?

Reply:
No, it's not ecom tokens.
```ts
wixEcomTokens = await client.auth.generateVisitorTokens();
```

Reply:
yeah, I agree. what name do you like more, `wixTokens`, `wixAuthTokens`, or something else?

Reply:
`wixVisitorTokens`

Reply:
If I understand correctly what they are.

> https://dev.wix.com/docs/sdk/core-modules/sdk/oauth-strategy#generatevisitortokens

They say: use this function when creating a session for visitors.

"Visitors" includes not logged in visitors. But if these tokens are not simply session tokens, but authentication tokens, then `wixAuthTokens`.

Reply:
they call anonymous users "visitors" and logged in users "members"

Reply:
Do we need to have these tokens for anonymous visitors as well?

Reply:
correct. because anonymous user may add products to a cart and even place an order. it is not required to log in for that. so, the session is generated for anonymous users as well.
from wix api we have `generateVisitorTokens` to create new tokens for anonymous user and `getMemberTokens` that finishes login process and generates tokens for logged in user

Reply:
Then `wixVisitorTokens` sounds like an appropriate name.

Reply:
I like `wixAuthTokens` better, WDYT?

Reply:
Didn't we just discussed that these are visitor tokens and not authentication tokens.

Reply:
Do we save them in our session under the same key both for logged in and anonymous users?

Reply:
Yes we do:

https://github.com/codux-templates/e-commerce-reclaim/blob/0d39cc8388d1913989e47b0e3c6fca81bec11670/lib/ecom/session.ts#L43-L44

So if you name them `wixAuthTokens` how does it make sense to set them for anonymous visitor?

Reply:
We also need to explain why we save `wixClientId` into the session. Because normally that doesn't make sense, it's not session-dependent information.

Reply:
what is a difference between `visitor` and `authentication` tokens in your eyes?
I mean, these tokens authenticates particular visitor for wix services

Reply:
https://dev.wix.com/docs/go-headless/coding/java-script-sdk/visitors-and-members/handle-visitors

> Once you've [created a client](https://dev.wix.com/docs/go-headless/coding/java-script-sdk/visitors-and-members/create-a-client-with-oauth), your client can generate, manage, and use tokens for anonymous site visitors. The client uses these tokens when making requests to Wix APIs on behalf of a visitor to maintain the visitor's session.


so, anonymus user is just an unknown user, but it has session and authorization

Reply:
> Then wixVisitorTokens sounds like an appropriate name.

@alisey anyone who will go to wix sdk docs will be confused, because docs say that visitor is an anonymous user, but code says that `wixVisitorTokens` is used for both anonymous and logged in users. 
maybe `wixSessionToken`?

Reply:
renamed to `wixSessionTokens `

Reply:
>We also need to explain why we save wixClientId into the session. Because normally that wouldn't make sense, it's not session-dependent information.

there already was a comment few lines above where we reading it. anyway, improved that comment, added one more at the line where we saving it to session